Content.Trauma.Shared/Botany/Components/SharedPlantAnalyzerComponent.cs

```diff
@@ -0,0 +1,144 @@
+using Content.Shared.Atmos;
+using Content.Shared.Botany.Components;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization;
+
+namespace Content.Trauma.Shared.Botany.Components;
+
+[Serializable, NetSerializable]
+public partial struct SeedChemQuantityHelper
+{
+    /// <summary>
+    /// Minimum amount of chemical that is added to produce, regardless of the potency
+    /// </summary>
+    [DataField("Min")] public int Min = 0;
+
+    /// <summary>
+    /// Maximum amount of chemical that can be produced after taking plant potency into account.
+    /// </summary>
+    [DataField("Max")] public int Max = 5;
+
+    /// <summary>
+    /// When chemicals are added to produce, the potency of the seed is divided with this value. Final chemical amount is the result plus the `Min` value.
+    /// Example: PotencyDivisor of 20 with seed potency of 55 results in 2.75, 55/20 = 2.75. If minimum is 1 then final result will be 3.75 of that chemical, 55/20+1 = 3.75.
+    /// </summary>
+    [DataField("PotencyDivisor")] public int PotencyDivisor = 20;
+
+    /// <summary>
+    /// Inherent chemical is one that is NOT result of mutation or crossbreeding. These chemicals are removed if species mutation is executed.
+    /// </summary>
+    [DataField("Inherent")] public bool Inherent = true;
+
+    public SeedChemQuantityHelper(int min, int max, int potencyDivisor, bool inherent)
+    {
+        Min = min;
+        Max = max;
+        PotencyDivisor = potencyDivisor;
+        Inherent = inherent;
+    }
+}
+
+
+public enum PlantAnalyzerModes
+{
+    Scan,
+    Extract,
+    Implant,
+    DeleteMutations
+}
+
+public enum HarvestType : byte
+{
+    NoRepeat,
+    Repeat,
+    SelfHarvest
+}
```

Comment:
put it in `Content.Trauma.Common.Botany` not shared so core files can access it
remove the original from content.server and update every usage of it to have a `using Content.Trauma.Common.Botany;`